fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -537,11 +541,17 @@ public Expression visitPredicated(PredicatedContext ctx) {
      * @return Expression
      */
     public Expression withPredicate(PredicateContext ctx, Expression e) {
+        boolean isNot = ctx.NOT() != null ? true : false;
+        Expression outExpression;
         switch (ctx.kind.getType()) {
             case DorisParser.BETWEEN:
-                return withBetween(ctx, e);
+                outExpression = withBetween(ctx, e);
+                return isNot ? new Not(outExpression) : outExpression;
+            case DorisParser.IN:
+                outExpression = withIn(ctx, e);
+                return isNot ? new Not(outExpression) : outExpression;
```

Comment:
we can move this statement to the end of this function and do not need process in each case

Reply:
done